crates/rust-analyzer/src/integrated_benchmarks.rs

```diff
@@ -0,0 +1,186 @@
+//! Fully integrated benchmarks for rust-analyzer, which load real cargo
+//! projects.
+//!
+//! The benchmark here is used to debug specific performance regressions. If you
+//! notice that, eg, completion is slow in some specific case, you can  modify
+//! code here exercise this specific completion, and thus have a fast
+//! edit/compile/test cycle.
+//!
+//! Note that "Rust Analyzer: Run" action does not allow running a single test
+//! in release mode in VS Code. There's however "Rust Analyzer: Copy Run Command Line"
+//! which you can use to paste the command in terminal and add `--release` manually.
+
+use std::{convert::TryFrom, sync::Arc};
+
+use ide::{Change, CompletionConfig, FilePosition, TextSize};
+use ide_db::helpers::{insert_use::InsertUseConfig, merge_imports::MergeBehavior, SnippetCap};
+use test_utils::project_root;
+use vfs::{AbsPathBuf, VfsPath};
+
+use crate::cli::load_cargo::{load_workspace_at, LoadCargoConfig};
+
+#[test]
+fn integrated_highlighting_benchmark() {
+    // Don't run slow benchmark by default
+    if true {
```

Comment:
Wait, how does this work?

Reply:
I uncomment this manually when I want to capture profile :-)

Reply:
Might be worth checking for an env variable instead.